optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-backend/src/main/java/org/optaplanner/workbench/screens/solver/backend/server/SolverValidator.java

```diff
@@ -16,38 +16,108 @@
 
 package org.optaplanner.workbench.screens.solver.backend.server;
 
-import java.util.ArrayList;
+import java.io.ByteArrayInputStream;
 import java.util.List;
 import javax.inject.Inject;
+import javax.inject.Named;
 
+import com.google.common.base.Charsets;
+import org.drools.compiler.kie.builder.impl.InternalKieModule;
+import org.drools.compiler.kie.builder.impl.KieContainerImpl;
+import org.drools.compiler.kie.builder.impl.KieModuleKieProject;
+import org.guvnor.common.services.backend.file.JavaFileFilter;
 import org.guvnor.common.services.shared.message.Level;
 import org.guvnor.common.services.shared.validation.model.ValidationMessage;
-import org.optaplanner.core.config.SolverConfigContext;
-import org.optaplanner.core.config.solver.SolverConfig;
+import org.kie.api.KieServices;
+import org.kie.api.runtime.KieContainer;
+import org.kie.workbench.common.services.backend.builder.LRUBuilderCache;
+import org.kie.workbench.common.services.backend.validation.asset.Validator;
+import org.kie.workbench.common.services.backend.validation.asset.ValidatorFileSystemProvider;
+import org.kie.workbench.common.services.shared.project.KieProject;
+import org.kie.workbench.common.services.shared.project.KieProjectService;
+import org.optaplanner.core.api.solver.SolverFactory;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.io.IOService;
 
 public class SolverValidator {
 
+    private IOService ioService;
+
+    private KieProjectService projectService;
+
     @Inject
-    private ConfigPersistence configPersistence;
+    private LRUBuilderCache builderCache;
 
-    public List<ValidationMessage> validate( String xml ) {
-        ArrayList<ValidationMessage> validationMessages = new ArrayList<ValidationMessage>();
+    public SolverValidator() {
+    }
 
-        try {
-            SolverConfig solverConfig = configPersistence.toSolverConfig( xml );
-            solverConfig.buildSolver( new SolverConfigContext() );
+    @Inject
+    public SolverValidator( final @Named( "ioStrategy" ) IOService ioService,
+                            final KieProjectService projectService ) {
+        this.ioService = ioService;
+        this.projectService = projectService;
+    }
 
-        } catch (Exception e) {
-            validationMessages.add( make( e ) );
+    public List<ValidationMessage> validate( final Path resourcePath,
+                                             final String content ) {
```

Comment:
resourcePath: PathImpl{uri='default:///home/.../optaplanner-wb/optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-backend/target/test-classes/ProjectA/src/main/resources/org/projectA/my.solver.xml', fileName='my.solver.xml', attrs={}}
content: the solver XML

There's no way that org.projectA.MySolution located in src/main/resources/ProjectA/src/main/resources/org/projectA/MySolution.java can end up in the correct in the kjar under just "org/projectA/".